pkg/endpointmanager/conntrack.go

```diff
@@ -79,51 +83,44 @@ func RunGC(e *endpoint.Endpoint, isLocal, isIPv6 bool, filter *ctmap.GCFilter) {
 	}
 }
 
+// skipGC determines whether we should skip running GC for this endpoint.
+func skipGC(e *endpoint.Endpoint) bool {
+	e.Mutex.RLock()
+	defer e.Mutex.RUnlock()
+
+	if e.SecurityIdentity == nil || !e.Opts.IsEnabled(option.ConntrackLocal) {
+		return true
+	}
+
+	return false
+}
+
 // EnableConntrackGC enables the connection tracking garbage collection.
 func EnableConntrackGC(ipv4, ipv6 bool) {
 	go func() {
-		seenGlobal := false
 		sleepTime := time.Duration(GcInterval) * time.Second
 		for {
 			eps := GetEndpoints()
+			if len(eps) > 0 {
+				if ipv4 {
+					RunGC(nil, true, ctmap.NewGCFilterBy(ctmap.GCFilterByTime))
+				}
+				if ipv6 {
+					RunGC(nil, false, ctmap.NewGCFilterBy(ctmap.GCFilterByTime))
```

Comment:
I think the logic of the `false` should be `true` and on line 106 it should be `false`.